StreetHawk/Classes/Core/Publish/SHUtils.m

```diff
@@ -863,44 +863,85 @@ - (NSString *)platformString
 
 @implementation UIColor (SHExt)
 
++ (BOOL)isRGB:(NSArray *)arrayComponents
+{
+    return arrayComponents.count >= RGB_COLOUR_CODE_LEN;
+}
+
++ (BOOL)isRGBA:(NSArray *)arrayComponents
+{
+    return arrayComponents.count == ARGB_COLOUR_CODE_LEN;
+}
+
 + (UIColor *)colorFromHexString:(NSString *)hexString
 {
     if (![hexString isKindOfClass:[NSString class]])
     {
         return nil;
     }
-    if (hexString.length != 4 //#RGB
-        && hexString.length != 7 //#RRGGBB
-        && hexString.length != 9) //AARRGGBB
-    {
-        return nil;
-    }
-    if (![hexString hasPrefix:@"#"])
+    CGFloat red = -1;
+    CGFloat green = -1;
+    CGFloat blue = -1;
+    CGFloat alpha = -1;
+    if (hexString.length == RGB_COLOUR_CODE_LEN + 1 //#RGB
+        || hexString.length == RRGGBB_COLOUR_CODE_LEN + 1 //#RRGGBB
+        || hexString.length == AARRGGBB_COLOUR_CODE_LEN + 1) //#AARRGGBB
     {
-        return nil;
-    }
-    if (hexString.length == 4)
-    {
-        NSString *red = [hexString substringWithRange:NSMakeRange(1, 1)];
-        NSString *green = [hexString substringWithRange:NSMakeRange(2, 1)];
-        NSString *blue = [hexString substringWithRange:NSMakeRange(3, 1)];
-        hexString = [NSString stringWithFormat:@"#%@%@%@%@%@%@", red, red, green, green, blue, blue];
-    }
-    NSScanner *scanner = [NSScanner scannerWithString:hexString];
-    [scanner setScanLocation:1]; // bypass '#' character
-    unsigned rgbValue = 0;
-    [scanner scanHexInt:&rgbValue];
-    CGFloat red, green, blue, alpha;
-    red = ((rgbValue & 0xFF0000) >> 16)/255.0;
-    green = ((rgbValue & 0xFF00) >> 8)/255.0;
-    blue = (rgbValue & 0xFF)/255.0;
-    if (hexString.length == 7)
-    {
-        alpha = 1.0;
+        if (![hexString hasPrefix:@"#"])
```

Comment:
extract function , Its not evident waht its doing here, so please extract into function with name that decribes what it does